api/v1beta2/awsmachine_types.go

```diff
@@ -197,6 +197,39 @@ type Ignition struct {
 	// +kubebuilder:default="2.3"
 	// +kubebuilder:validation:Enum="2.3";"3.0";"3.1";"3.2";"3.3";"3.4"
 	Version string `json:"version,omitempty"`
+
+	// IgnitionProxy defines proxy settings for Ignition.
+	// Only valid for Ignition versions 3.4 and above.
+	// +optional
+	Proxy *IgnitionProxy `json:"proxy,omitempty"`
+
+	// TLS defines TLS settings for Ignition.
+	// Only valid for Ignition versions 3.4 and above.
+	// +optional
+	TLS *IgnitionTLS `json:"tls,omitempty"`
+}
+
+// IgnitionTLS defines TLS settings for Ignition.
+type IgnitionTLS struct {
+	// CASources defines the list of certificate authorities to use for Ignition.
+	// THe value is the certificate bundle (in PEM format). The bundle can contain multiple concatenated certificates.
+	// Supported schemes are http, https, tftp, s3, arn, gs, and `data` (RFC 2397) URL scheme.
+	//
+	// +optional
+	CASources []string `json:"certificateAuthorities,omitempty"`
+}
+
+// IgnitionProxy defines proxy settings for Ignition.
+type IgnitionProxy struct {
+	// HTTPProxy is the HTTP proxy to use for Ignition.
+	// +optional
+	HTTPProxy *string `json:"httpProxy,omitempty"`
+	// HTTPSProxy is the HTTPS proxy to use for Ignition.
+	// +optional
+	HTTPSProxy *string `json:"httpsProxy,omitempty"`
+	// NoProxy is the list of domains to not proxy for Ignition.
+	// +optional
+	NoProxy []string `json:"noProxy,omitempty"`
```

Comment:
This is a list of URLs right? Would be good to add validation for that and to explain in the godoc

Reply:
Not just URLs:

```
no_proxy (list of strings): specifies a list of strings to hosts that should be excluded from proxying. Each value is represented by an IP address prefix (1.2.3.4), an IP address prefix in CIDR notation (1.2.3.4/8), a domain name, or a special DNS label (*). An IP address prefix and domain name can also include a literal port number (1.2.3.4:80). A domain name matches that name and all subdomains. A domain name with a leading . matches subdomains only. For example foo.com matches foo.com and bar.foo.com; .y.com matches x.y.com but not y.com. A single asterisk (*) indicates that no proxying should be done.
```

I haven't opted to add much validation here because these values are a bit free-form and special cased. It'd be great to rely on butane itself to validate these in the webhook

Reply:
While yes, there's some flexibility to this string value, I think you could reasonably validate that with CEL expressions.

If this were me, I'd be trying something like below, but, some of this CEL isn't available until 1.31 so we may have to rely on a webhook for now
```
// +kubebuilder:validation:MaxItems=64
// +kubebuilder:vadliation:XValidation:rule="self == '*' || isIP(self) || isCIDR(self) || self.matches('^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$' || self.contains(':') && isIP(self.split(':')[0]) && self.split(':')[1].matches('\d+')", message="..."
```

It would be helpful to either include that string explaining the valid values, or link to a reference so that it's documented what we expect.